devito/operator.py

```diff
@@ -236,6 +236,7 @@ def _add_implicit(self, expressions):
                 try:
                     dims = [d.root for d in e.free_symbols if isinstance(d, Dimension)]
                     sub_dims = [d.root for d in e.subdomain.dimensions]
+                    sub_dims.append(e.subdomain.implicit_dimension)
```

Comment:
Is there an MFE this fixes to go as a test?

Reply:
fair point

Reply:
Yes, a shorter version of #827 would do - although it would only fail ~50% of the time (depending on how `free_symbols` works).